### ISSUE TITLE
Create a level_type in the condition struct to limit actions with player level

### ISSUE THREAD

**Issue by WPDolan** [uchicago-cs/chiventure#929]
Continuing from issue #880, we would like to update existing functionality in the condition module to restrict actions based off of fields that have been added into the player struct. We decided that the new fields warrant the creation of new corresponding condition types in `condition.c`. 

The player interface does not currently support functionality for accessing many of its new fields. Until functionality has been added to interact with all relevant fields, we are limited to adding condition types which are accessible through the current interface.

Functions currently exist within the player module to access and modify the player's level. We will use this implementation to create a corresponding `level_type` in `condition.c` with corresponding functions to check if the player is a certain level when attempting an action. This should provide the foundation for future issues that create condition types for other fields in the player struct once the interface is fully implemented.

**Comment by WPDolan** [comment]
I have created the `level_condition_t` condition type in `condition.h`, and I updated the existing condition struct to accommodate the new condition type. 

I have also created three new functions to handle level type conditions:
-  `level_condition_new` creates the new level type conditions. It takes an integer for the minimum level and a pointer to a player whose level will be checked.
- `check_level_condition` checks if the player's level meets the level requirement
- `valid_level_condition` ensures that the `level_condition_t` points to an existing player 

Fortunately, the existing functions in `condition.c` allowed for a straightforward integration of level conditions; I only needed to add pathways to the switch statements of the existing `check_condition`, `valid_condition`, and `free_condition` functions to fully integrate level type conditions within the condition module.
